src/sentry/tsdb/inmemory.py

```diff
@@ -27,42 +27,51 @@ def __init__(self, *args, **kwargs):
         super(InMemoryTSDB, self).__init__(*args, **kwargs)
         self.flush()
 
-    def incr(self, model, key, timestamp=None, count=1):
+    def incr(self, model, key, timestamp=None, count=1, environment=None):
+        environments = set([environment, None])
+
         if timestamp is None:
             timestamp = timezone.now()
 
         for rollup, max_values in six.iteritems(self.rollups):
             norm_epoch = self.normalize_to_rollup(timestamp, rollup)
-            self.data[model][key][norm_epoch] += count
+            for environment in environments:
+                self.data[model][(key, environment)][norm_epoch] += count
+
+    def merge(self, model, destination, sources, timestamp=None, environments=None):
+        environments = (set(environments) if environments is not None else set()).union([None])
```

Comment:
This could be factored out but I couldn't think of a descriptive name for it so I didn't yet.